test/parallel/test-http2-client-settings-errors.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const {
+  constants,
+  Http2Session,
+  nghttp2ErrorString
+} = process.binding('http2');
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const http2 = require('http2');
+
+// tests error handling within requestOnConnect
+// - NGHTTP2_ERR_NOMEM (should emit session error)
+// - every other NGHTTP2 error from binding (should emit session error)
+
+const specificTestKeys = [
+  'NGHTTP2_ERR_NOMEM'
+];
+
+const specificTests = [
+  {
+    ngError: constants.NGHTTP2_ERR_NOMEM,
+    error: {
+      code: 'ERR_OUTOFMEMORY',
+      type: Error,
+      message: 'Out of memory'
+    }
+  }
+];
+
+const genericTests = Object.getOwnPropertyNames(constants)
+  .filter((key) => (
+    key.indexOf('NGHTTP2_ERR') === 0 && specificTestKeys.indexOf(key) < 0
+  ))
+  .map((key) => ({
+    ngError: constants[key],
+    error: {
+      code: 'ERR_HTTP2_ERROR',
+      type: Error,
+      message: nghttp2ErrorString(constants[key])
+    }
+  }));
+
+const tests = specificTests.concat(genericTests);
+
+runTest(tests.shift());
+
+function runTest(test) {
+  const server = http2.createServer(common.mustNotCall());
```

Comment:
I might be missing something but is there a reason the server needs to be started on each test run rather than having a single server for the whole test?

Reply:
The `sessionError` is thrown on the server.
A single server was used originally, but I was not able to override the error thrown on that server. So a new server was created per test.

Reply:
Would it be possible to just do `server.once('sessionError', fn)` inside `runTest` and have all the code around creating a server and initiating it outside of `runTest`? If not, no worries.

Reply:
`server.once('sessionError', fn)` worked!
Thanks for the suggestion. I've updated the commit.